src/persist-client/examples/persistcli.rs

```diff
@@ -65,6 +65,14 @@ fn main() {
     let root_span = info_span!("persistcli");
     let res = match args.command {
         Command::Maelstrom(args) => runtime.block_on(async move {
+            // Persist internally has a bunch of sanity check assertions. If
+            // maelstrom tickles one of these, we very much want to bubble this
+            // up into a process exit with non-0 status. It's surprisingly
+            // tricky to be confident that we're not accidentally swallowing
+            // panics in async tasks (in fact there was a bug that did exactly
+            // this at one point), so abort on any panics to be extra sure.
+            mz_ore::panic::set_abort_on_panic();
```

Comment:
To get CI to pass, you'll also need to copy the scan change in https://github.com/MaterializeInc/materialize/pull/13952/commits/910ede7d9cc1747e05ff8b2d7061cce3eb3b3cfa